builtin/providers/docker/data_source_docker_image.go

```diff
@@ -0,0 +1,78 @@
+package docker
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/heroku/docker-registry-client/registry"
+)
+
+func dataSourceDockerImage() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDockerImageRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"id": &schema.Schema{
```

Comment:
In case you'll have any issues with using `d.SetId()` for an actual image hash I would suggest adding a new field called `digest` or `sha256_digest` as that's what it's called in the latest Docker API schema.